### PR TITLE
Track passing/failing tests in fiber

### PR DESCRIPTION
**Pull request by spicyj**
Run scripts/fiber/record-tests to re-record, then check git diff to see what you changed.

### CODE REVIEW

File: scripts/jest/test-framework-setup.js [hunk 1]
```diff
@@ -3,6 +3,12 @@
 // We want to globally mock this but jest doesn't let us do that by default
 // for a file that already exists. So we have to explicitly mock it.
 jest.mock('ReactDOM');
+jest.mock('ReactDOMFeatureFlags', () => {
+  const flags = require.requireActual('ReactDOMFeatureFlags');
+  return Object.assign({}, flags, {
+    useFiber: !!process.env.REACT_DOM_JEST_USE_FIBER,
+  });
+});
```
Comment (sebmarkbage) [inline scripts/jest/test-framework-setup.js]: Don't we somehow have a flag already?

**Comment by spicyj** [comment]
The other one failed on my machine too.

File: scripts/jest/test-framework-setup.js (hunk 1 above)
Comment (spicyj) [inline scripts/jest/test-framework-setup.js]: Yeah but it requires changing ReactDOMFeatureFlags.js which feels safe on Travis and not as safe if you might have local changes.

**Comment by sebmarkbage** [comment]
ok

File: package.json [hunk 2]
```diff
@@ -59,6 +59,9 @@
     "gulp-util": "^3.0.7",
     "gzip-js": "~0.3.2",
     "jest": "^15.1.1",
+    "jest-cli": "^15.1.1",
```
Comment (keyanzhang) [inline package.json]: Is the `jest` package just an alias for `jest-cli`? Only one is needed I think.

Comment (spicyj) [inline package.json]: jest seemingly just delegates to jest-cli but it does have the same exports so (y).

File: .travis.yml [hunk 3]
```diff
@@ -94,6 +94,13 @@ script:
       node scripts/facts-tracker/index.js \
         "fiber-tests" "$FIBER_TESTS"
 
+      # Should consolidate this with the above fiber run
+      echo 'Testing in fiber mode against test records...'
+      scripts/fiber/record-tests
+      git --no-pager diff scripts/fiber
+      FIBER_TESTS_STATUS=$(git status --porcelain scripts/fiber)
+      test -z "$FIBER_TESTS_STATUS"
```
Comment (sebmarkbage) [inline .travis.yml]: Does this mean that any time anyone adds a passing test, unrelated to Fiber, in a PR we will get a Travis error on the PR?

Comment (spicyj) [inline .travis.yml]: That was my intention. I figured it was good to know whether your new test passes in Fiber. Not good?

Comment (sebmarkbage) [inline .travis.yml]: It's good for the team but might be noisy for third-party PR. Not too worried about it though. Let's see how it works out.

**Comment by spicyj** [comment]
@Aweary Heads up if you didn't see: everyone needs to run scripts/fiber/record-tests when adding/removing tests (or breaking/fixing them in the fiber reconciler) or else CI will fail now. This way it's really obvious whether tests pass or fail in fiber when looking at a diff.

**Comment by Aweary** [comment]
@spicyj thanks, I didn't see this. I'll make sure to do this 👍

**Pull request merged**